### DISCUSSION TITLE
Which TextTrack API

### DISCUSSION THREAD

**Discussion by markstevening** [videojs/video.js#8858] [Q&A] [answered]
Hi,

Is there supposed to be a difference between using 

player.addTextTrack(kind, label, language);
vs
player.textTracks().addTrack(new TextTrack()) ?

Is one approach recommended over another?  When experimenting with generating text tracks on the fly, I was having trouble with getting them to display when using the second method, and they seem to result in different objects being created in the list of text tracks. 

Is there a reason the player.addTextTrack doesn't receive a TextTrack object, or is it a historical curiosity?

Thanks

**Comment by gkatsev** [comment]
You should use the player API. It's designed to look like the native API on the video element. You also may want to use `addRemoteTextTrack`, if you want to be able to remove these tracks because `addTextTrack` doesn't guarantee it if you may be using native text tracks (which is the default for safari).
When you write `new TextTrack()` which text track object are you referring to?

**Comment by markstevening** [comment]
I was using the player.addTextTrack() method, but I had to change to the other approach, as I need to set the .id on the track object, which doesn't seem to be possible when using the first method.

`
import {
  VidstackPlayer,
  VidstackPlayerLayout,
  TextTrack,
} from "https://cdn.vidstack.io/player";

// ...

const playerTrack = new TextTrack({
  id: id,
  kind: kind,
  mode: mode,
  label: label,
  language: language,
});

player.textTracks().addTrack(playerTrack);

`

This seems to work well for kind : "chapters",  but "captions" tracks don't seem to be displayed in the player.

**Reply by gkatsev** [reply, marked as answer]
Your code example is using vidstack not Video.js. If you're actually using vidstack, you should take a look at their documentation for API usage.

**Reply by markstevening** [reply]
Oh no, it's even worse than that. I'm an **idiot**. I was experimenting testing both of them, and accidentally left in some code for vidstack when I switched to videojs.

It worked partly (presumably some API overlap), and I didn't even notice I was using TextTrack from the wrong project instead of https://docs.videojs.com/texttrack.

Thanks
